include/zephyr/drivers/gpio.h

```diff
@@ -129,6 +129,16 @@ extern "C" {
  */
 #define GPIO_INT_HIGH_1                (1U << 26)
 
+#ifdef CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT
+/* Disable/Enable interrupt functionality without changing other interrupt
+ * related register, such as clearing the pending register.
+ *
+ * This is a component flag that should be combined with `GPIO_INT_ENABLE` or
+ * `GPIO_INT_DISABLE` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_ENABLE_DISABLE_ONLY   (1u << 27)
```

Comment:
Please also update the documentation for GPIO [interrupt configuration flags](https://github.com/zephyrproject-rtos/zephyr/blob/e2742af6ed1118e1e2dd2b0ffe89ce5c210ab22d/include/zephyr/drivers/gpio.h#L82)  to provide details about the behavior of pending interrupts.

Something like:
"The GPIO controller should clear any pending interrupt on the pin when changing the triggering properties.  Applications should use the GPIO_INT_MODE_DISABLE_ONLY and GPIO_INT_MODE_ENABLE_ONLY flags to enable and disable interrupts on the pin without changing any GPIO settings."